src/SessionSetup/RetrieveInstancesWidgetTest.cpp

```diff
@@ -449,4 +461,51 @@ TEST_F(RetrieveInstancesWidgetTestStarted, ProjectChangeFailed) {
   // "Test Project 2"
 }
 
+TEST_F(RetrieveInstancesWidgetTestStarted, AllCheckboxSuccessful) {
+  EXPECT_CALL(mock_retrieve_instances_,
+              LoadInstances(std::optional<Project>(std::nullopt),
+                            InstanceListScope(InstanceListScope::kAllReservedInstances)))
+      .WillOnce(Return(Future<ErrorMessageOr<QVector<Instance>>>(kTestInstancesProject1All)));
+  EXPECT_CALL(mock_retrieve_instances_,
+              LoadInstances(std::optional<Project>(std::nullopt),
+                            InstanceListScope(InstanceListScope::kOnlyOwnInstances)))
+      .WillOnce(Return(Future<ErrorMessageOr<QVector<Instance>>>(kTestInstancesProject1)));
+
+  QTest::mouseClick(all_check_box_, Qt::MouseButton::LeftButton);
+  QCoreApplication::processEvents();
+
+  VerifyLastLoadingReturnedInstanceList(kTestInstancesProject1All);
+  VerifyAndClearSignalsOfSuccessfulLoadingCycle();
+  VerifyAllElementsAreEnabled();
```

Comment:
Here we are checking 3 methods twice. In previous methods we are checking a subset of them. It's not always very clear the difference. What do you think about having a helper Verify function that could take a few parameters and based on that execute the right Verify methods? Could make these tests easy to understand and to modify.

Reply:
I encapsulated these 3 calls into `VerifySuccessfulLoadAndUIState` and used it where applicable in the code.